app/scripts/model/dataProxy.js

```diff
@@ -254,12 +254,16 @@
 
                         // --- web API results converting ----
                         var _patientData = [], _sampleData = [];
-                        var _patientIds = _.uniq(_.pluck(_ajaxPatientData, 'patient_id'));
-                        var _sampleIds = _.uniq(_.pluck(_ajaxSampleData, 'sample_id'));
+                        var _patientIds = [];
                         var _indexSample = 0, _sampleDataIndicesObj = {};
                         var _storedCnaGeneInventory = {}, _storedCnaGeneIndex = 0;
                         var _storedMutGeneInventory = {}, _storedMutGeneIndex = 0;
-                        _.each(_sampleIds, function (_sampleId) {
+                        _.each(Object.keys(_ajaxPatient2SampleIdMappingObj), function (_tmpPatientId) {
+                          _patientIds.push(_tmpPatientId);
+                          var _sampleIdArr = _ajaxPatient2SampleIdMappingObj[_tmpPatientId];
+                          _.each(_sampleIdArr, function (_sampleId) {
+                            // map from sample to patient
+                            _ajaxSample2PatientIdMappingObj[_sampleId] = new Array(_tmpPatientId);
```

Comment:
Can we keep the consistency to use [_tmpPatientId] and avoid new Array or new Object?